### ISSUE TITLE
The vertical scroll is under the backdrop.

### ISSUE THREAD

**Issue by pawelchmara** [biig-io/ngx-smart-modal#162]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
The vertical scroll is hidden by the backdrop when the modal is higher then page and you cannot grab it and move the page. When you click on the backdrop, the modal closes.

**Expected behavior**
The vertical scroll should by on the top of DOM element stack.

**Reproduction of the problem**
![scroll_problem](https://user-images.githubusercontent.com/20394260/48776839-2f426300-ecd1-11e8-9fca-a2e17fadc93b.png)

**Please tell us about your environment:**

* **Smart Modal version:**  7.1.0 (The problem also occured while using 6.x.x version of NgsSmartModal)

* **Angular version:** 6.0.7

* **Browser:** all

* **Language:** Typescript 2.7.2

* **Bootstrap :** 4.1.1.

**Comment by maximelafarie** [comment]
Unless you provide a demo of your issue (Stackblitz preferred), I think it's a normal behavior of the library, please tell me if you got the same behavior that on the [demo](https://biig-io.github.io/ngx-smart-modal/) (please click on "big content modal button").

**Comment by pawelchmara** [comment]
Hi,

I the demo example vertical scroll works great. However I have noticed that ngx-smart-modal DOM element is rendered in different way in my app and demo app. Please look at an example:

1. Demo example: nsm-dialog is inside nsm-overlay-open
![1](https://user-images.githubusercontent.com/20394260/48823667-f69e9a00-ed61-11e8-8572-0ff684e1c08c.png)

2. My example: nsm-dialog and nsm-overlay-open are at the same level:
![2](https://user-images.githubusercontent.com/20394260/48823688-0fa74b00-ed62-11e8-9705-c0a01a90cc8f.png)

Maybe do you know way? What could be the reason for this behavior? 

I know that without full example will be hard to answer my question, but I try to show how I use ng-smart-modal in my app.

Template file:

```
<ngx-smart-modal
  #jobDialog
  [identifier]="id"
  [closable]="false"
  (onClose)="onClose()"
  (onOpen)="onOpen()"
  (onDataAdded)="onDataAdded()"
  (onCloseFinished)="onCloseFinish()"
>
  <div class="modal-header">
    <h4 class="modal-title">
      Ttitle
    </h4>
    <button type="button" class="close" data-dismiss="modal" aria-hidden="true" (click)="jobDialog.close()">&times;</button>
  </div>
  <div class="modal-body">
    <div class="row mb-4">
      <div class="col-12 col-xl-8 mb-2 mb-xl-0">
        ...
      </div>
      <div class="col-12 col-xl-4">
        ...
      </div>
    </div>

    <div class="row">
      <div class="col-12 col-md-6 col-xl-3 mb-2 mb-xl-0">
        ...
      </div>
      <div class="col-12 col-md-6 col-xl-3 mb-2 mb-xl-0">
        ...
      </div>
      <div class="col-12 col-md-6 col-xl-3 mb-2 mb-xl-0">
        ...
      </div>
      <div class="col-12 col-md-6 col-xl-3 mb-2 mb-xl-0">
        ...
      </div>
    </div>
  </div>
  <div class="modal-footer">
    <button type="button" class="btn btn-secondary" data-dismiss="modal" (click)="jobDialog.close()">
      <fa name="check" [fixed]="true"></fa><span>Ok</span>
    </button>
  </div>
</ngx-smart-modal>
```

Component file:

Get reference to modal:
`@ViewChild('jobDialog') dialog: NgxSmartModalComponent;`

Open modal on observable subscribtion always after ngAfterViewInit:
`this.dialog.open();`

Thx!

**Comment by maximelafarie** [comment]
@pawelchmara Hi, please excuse me for the late answer.

If you haven't already fixed your issue, I suggest you to use the `NgxSmartModalService` `open()` method to open a modal remotely instead of using `@ViewChild()`. But it might work as well.

Are you really using version `7.1.0`? It seems strange because your DOM looks like of our version `6.x.x`.

I stay tuned! 🖖

**Comment by maximelafarie** [comment]
Hi @pawelchmara and thank you for your patience.

You can unblock the scroll by overriding default `.dialog-open` class in your top-level style file.

I made a live example here: https://stackblitz.com/edit/ngx-smart-modal-no-scroll-blocking

As I put this issue in our backlog, it may be useful to add an option to avoid blocking scroll if a modal is opened. Not a big deal, we have to see if it may be useful. 😉 

Regards! 🖖

**Comment by pawelchmara** [comment]
Super! Thank you very much for help.

**Comment by maximelafarie** [comment]
Fixed as a new SCSS variable has been created to customize this option easily. Available very soon in `7.2.0`! 🔥 
Regards! 🖖
